application/models/file.go

```diff
@@ -16,6 +16,12 @@ import (
 	"github.com/gofrs/uuid"
 )
 
+type FileUploadError struct {
```

Comment:
I like this implementation. I've never created a custom error type, but I'm curious why this doesn't have a `Error()` method.

Reply:
Funny enough, I wasn't thinking of it as implementing `error`.  I just gave it the attributes I wanted to be able to pull out of it.  I guess I could have given it and `Error()` method that just returns the `Message`. Do you think that's worth doing?

Reply:
Seems like it would make sense. Fewer lines affected maybe.